### PR TITLE
Robustify TLS nginx settings

### PR DESCRIPTION
**Pull request by mrgnr**
- Prevent [POODLE](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2014-3566) attacks by dropping SSL3 by default.
- Prevent [attacks against weak Diffie-Hellman keys](https://weakdh.org) by disabling export ciphers and using a 2048-bit Diffie-Hellman group.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @mrgnr! I've created [OSPR-1730](https://openedx.atlassian.net/browse/OSPR-1730) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



If you like, you can add yourself to the [AUTHORS](https://github.com/edx/configuration/blob/master/AUTHORS)      file for this repo, though that isn't required.     Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

File: playbooks/roles/nginx/tasks/main.yml [hunk 1]
```diff
@@ -1,6 +1,26 @@
 # requires:
 #   - common/tasks/main.yml
 ---
+
+- name: Create Diffie-Hellman parameters to prevent weak key exchange
+  command: openssl dhparam -out "{{ NGINX_DH_PARAMS_PATH | basename }}" 2048
```
Comment (gsong) [inline playbooks/roles/nginx/tasks/main.yml]: Why not make the key size configurable, defaulting to 2048 or 4096?

Comment (mrgnr) [reply playbooks/roles/nginx/tasks/main.yml]: It's now configurable.

**Comment by mrgnr** [comment]
@gsong Are these changes good to merge?

**Comment by gsong** [comment]
@mrgnr Not yet, @edx/devops still needs to review.

**Review by jibsheet — APPROVED** [review]

**Review by feanil — APPROVED** [review]

**Pull request merged**